jobs/jobs_api_go_sdk.go

```diff
@@ -174,21 +218,21 @@ func updateJobClusterSpec(clusterSpec *compute.ClusterSpec, d *schema.ResourceDa
 
 func prepareJobSettingsForUpdateGoSdk(d *schema.ResourceData, js *JobSettingsResource) error {
 	if js.NewCluster != nil {
-		err := updateJobClusterSpec(js.NewCluster, d)
+		err := updateJobClusterSpec(d, "new_cluster.0", js.NewCluster)
 		if err != nil {
 			return err
 		}
 	}
-	for _, task := range js.Tasks {
-		if task.NewCluster != nil {
-			err := updateJobClusterSpec(task.NewCluster, d)
+	for i := range js.Tasks {
+		if js.Tasks[i].NewCluster != nil {
+			err := updateJobClusterSpec(d, fmt.Sprintf("task.%d.new_cluster.0", i), js.Tasks[i].NewCluster)
 			if err != nil {
 				return err
 			}
 		}
 	}
 	for i := range js.JobClusters {
-		err := updateJobClusterSpec(&js.JobClusters[i].NewCluster, d)
+		err := updateJobClusterSpec(d, fmt.Sprintf("job_cluster.%d.new_cluster.0", i), &js.JobClusters[i].NewCluster)
```

Comment:
These changes were necessary for adding a `schema.ResourceData` prefix argument.